app/services/import/companies.rb

```diff
@@ -17,7 +17,14 @@ def call
 
     def import
       import_each_with_logging(csv, FILEPATH) do |row|
-        company = Company.find_or_initialize_by(isin: row[:isin])
+        company = Company.find_or_initialize_by(name: row[:company_name])
+
+        company.isin = if company.isin.present?
```

Comment:
couldn't this be moved to `company_attributes` method?

Reply:
probably so.  I kept it here, but it doesn't make so much sense now. I'll move it.

Reply:
hmmm, actually, I would need to pass the current isin to that method if I were to manage this there. Currently the `company_attributes` is not aware of the record being updated. So not sure if that's worth it.